operator/pkg/apis/operator/v1alpha1/type.go

```diff
@@ -0,0 +1,771 @@
+	// +optional
+	KubernetesVersion string `json:"kubernetesVersion,omitempty"`
+
+	// Conditions represents the latest available observations of a karmada's current state.
+	// +optional
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+}
+
+// LocalSecretReference is a reference to a secret within the enclosing
+// namespace.
+type LocalSecretReference struct {
+	// Namespace is the namespace for the resource being referenced.
+	Namespace string `json:"namespace,omitempty"`
+
+	// Name is the name of resource being referenced.
+	Name string `json:"name,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// KarmadaList is a list of Karmadas.
+type KarmadaList struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// +optional
+	metav1.ListMeta `json:"metadata,omitempty"`
+
+	Items []Karmada `json:"items"`
+}
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:subresource:status
+// +kubebuilder:path=karmadaonborad,scope=Namespaced,categories={karmada-io}
+// +kubebuilder:printcolumn:JSONPath=`.spec.syncMode`,name="Mode",type=string
+// +kubebuilder:printcolumn:JSONPath=`.metadata.creationTimestamp`,name="Age",type=date
+
+// KarmadaOnborad enables declarative join member cluster to karmada.
+type KarmadaOnborad struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// Standard object's metadata.
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines the desired behavior of the KarmadaOnborad.
+	// +optional
+	Spec KarmadaOnboradSpec `json:"spec,omitempty"`
+
+	// Most recently observed status of the KarmadaOnborad.
+	// +optional
+	Status KarmadaOnboradStatus `json:"status,omitempty"`
+}
+
+// ClusterSyncMode describes the mode of synchronization between member cluster and karmada control plane.
+type ClusterSyncMode string
+
+const (
+	// Push means that the controller on the karmada control plane will in charge of synchronization.
+	// The controller watches resources change on karmada control plane then pushes them to member cluster.
+	Push ClusterSyncMode = "Push"
+
+	// Pull means that the controller running on the member cluster will in charge of synchronization.
+	// The controller, as well known as 'agent', watches resources change on karmada control plane then fetches them
+	// and applies locally on the member cluster.
+	Pull ClusterSyncMode = "Pull"
+)
+
+// KarmadaOnboradSpec defines the desired behavior of the KarmadaOnborad.
+type KarmadaOnboradSpec struct {
+	// SyncMode describes how a cluster sync resources from karmada control plane.
+	// +kubebuilder:validation:Enum=Push;Pull
+	// +required
+	SyncMode ClusterSyncMode `json:"syncMode"`
+
+	// ClusterName describes the karmada cluster resource name.
+	// +required
+	ClusterName string `json:"clusterName"`
+
+	// ClusterNamespace define namespace in the control plane where member secrets cluster are stored.
```

Comment:
ClusterNamespace defines the namespace in karmada control plane where secrets of member clusters are stored.